kernel/src/fs/manager.rs

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::sync::{Arc, Weak};
+
+use aster_systree::{
+    impl_cast_methods_for_branch, Error, Result, SysAttrSet, SysBranchNode, SysBranchNodeFields,
+    SysNode, SysNodeId, SysNodeType, SysObj, SysPerms, SysStr,
+};
+use inherit_methods_macro::inherit_methods;
+use ostd::mm::{VmReader, VmWriter};
+use spin::Once;
+
+/// Returns a reference to the [`FsManager`]. Panics if not initialized.
+pub fn singleton() -> &'static Arc<FsManager> {
+    MANAGER.get().expect("Fs manager has not been initialized")
+}
+
+/// Initializes the [`FsManager`] singleton
+pub fn init() {
+    MANAGER.call_once(FsManager::new);
+    aster_systree::singleton()
+        .root()
+        .add_child(singleton().clone())
+        .expect("Failed to add fs manager to SysTree");
+}
+
+static MANAGER: Once<Arc<FsManager>> = Once::new();
+
+/// Manager for virtual filesystem.
```

Comment:
All Rust docs should be written in plain and correct English. "filesystem" is invalid; "file system" is.

Try establish a good habit of using _articles_ ("a", "an", "the") correctly. I found this PR is careless on articles. For example, add "the" before "manager" because there is going to be one single manager.

```rust
/// The system-wide manager for file systems.
```

As another example, add "a" before "trait".

```rust
/// A trait to represent a filesystem's control interface.
pub trait FsFactory: SysObj {
```